cnf-certification-test/accesscontrol/suite.go

```diff
@@ -401,53 +402,81 @@ func testPodServiceAccount(env *provider.TestEnvironment) {
 //nolint:dupl
 func testPodRoleBindings(env *provider.TestEnvironment) {
 	ginkgo.By("Should not have RoleBinding in other namespaces")
-	failedPods := []string{}
+	var compliantObjects []*testhelper.ReportObject
+	var nonCompliantObjects []*testhelper.ReportObject
 
 	for _, put := range env.Pods {
 		ginkgo.By(fmt.Sprintf("Testing role binding for pod: %s namespace: %s", put.Name, put.Namespace))
-		if put.Spec.ServiceAccountName == "" {
-			logrus.Infof("%s has an empty serviceAccountName, skipping.", put.String())
+		if put.IsUsingDefaultServiceAccount() {
+			logrus.Infof("%s has an empty or default serviceAccountName, skipping.", put.String())
 			continue
```

Comment:
I don't think you should filter out pods that use the "default" ServiceAccount. It's ok for pods to use it and create rolebinding/clusterrolebindings that have it as the subject for the roles/clusterroles in order to grant access to the cluster api. Good example here https://stackoverflow.com/questions/52995962/kubernetes-namespace-default-service-account

Maybe my previous comment was a bit misleading. I just wanted to say that comparing with the empty string "" seemed redundant to me, as all the pods will come with, at least, the "default" serviceAccountName set. But the "default" is actually a valid SA that can be used by role/clusterrole-bindings.

Reply:
Ah gotcha.  Okay yes I will remove that check.